modules/nerd-fonts/nerd-fonts.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+set -oue pipefail
+
+NERD_FONTS_DIR=/usr/share/fonts/nerd-fonts
+COMPRESS_TYPES=("tar.xz" "zip")
+
+mkcd() {
+
+    if [ -d "$1" ]; then
+        
+        rm -rf "$1"
+
+    fi
+    
+    mkdir -p "$1"
+    cd "$1" || exit 0
+
+}
+
+download() {
+
+    local downloaded
+    local file
+
+    mkcd "$1"
+
+    echo "${COMPRESS_TYPES[@]}" | while IFS=$'\n' read -r -d ' ' type; do
+
+        downloaded=$(ls | wc -l)
+        file="$1.$type"
+
+        if [[ $downloaded -eq 0 ]]; then
+
+            echo "--- Downloading $file ---"
+
+            curl -o "$file" -OL https://github.com/ryanoasis/nerd-fonts/releases/latest/download/"$file"
+
+            if [[ -f "$file" ]]; then
+
+                case $type in
+
+                tar.xz) tar xvJf "$file" ;;
+                zip) unzip "$file" ;;
+
+                esac
+
+                rm -rf "$file"
+
+                echo "--- $file downloaded ---"
+
+            else
+
+                echo "--- Unable to download $file ---"
+
+            fi
+
+        fi
+
+    done
+
+    cd $NERD_FONTS_DIR
+
+}
+
+mkcd $NERD_FONTS_DIR
+
+echo "--- Downloading selected nerd-fonts ---"
+
+get_yaml_array nerdfonts '.fonts[]' "$1"
+
+for font in "${nerdfonts[@]}"; do
+
+    download "$(echo "$font" | tr -d '\n')"
+
+done
+
+fc-cache -f $NERD_FONTS_DIR
+
+echo "--- Nerd-fonts installed ---"
+
+cd /
```

Comment:
I don’t think this cd is necessary, build.sh should reset back to the config directory or smth like that when running each module. And if it doesn’t currently, that is an oversight by me and should be added in.

Reply:
I preferred to place it to ensure that the execution of other modules would not fail.

Reply:
That would be the job of build.sh, I will investigate further once I get the time.